rholang/src/main/scala/coop/rchain/rholang/interpreter/SystemProcesses.scala

```diff
@@ -68,59 +68,85 @@ object SystemProcesses {
 
       type Channels = Seq[Result[ListParWithRandomAndPhlos]]
 
+      type Producer = (Seq[Par], Par) => F[Unit]
+
       private val prettyPrinter = PrettyPrinter()
 
       private val UNLIMITED_MATCH_PHLO = matchListPar(Cost(Integer.MAX_VALUE))
 
-      private def foldResult(
-          produceResult: Either[InterpreterError, Option[(ContWithMetaData, Channels)]]
-      ): F[Unit] =
-        produceResult.fold(
-          _ => F.raiseError(OutOfPhlogistonsError),
-          _.fold(F.unit) {
-            case (cont, channels) =>
-              dispatcher.dispatch(unpackCont(cont), channels.map(_.value), cont.sequenceNumber)
+      object ContractCall {
```

Comment:
this object is very hard to read. I guess moving the definitions past their usages would help somewhat. Also, the nested `def produce` is quite surprising. I think a `val producer: Producer = (values: Seq[Par], channel: Par)) => ...` would be better

Reply:
This is cleaned up a little bit in the next commit